File: ObjectClass/object.java
public class Student {
    private String name;
    private int id;
    private double gpa;
    
    // Constructor
    public Student(String name, int id, double gpa) {
        this.name = name;
        this.id = id;
        this.gpa = gpa;
    }
    
    // Getter methods
    public String getName() { return name; }
    public int getId() { return id; }
    public double getGPA() { return gpa; }
    
    // Setter methods
    public void setGPA(double newGPA) {
        this.gpa = newGPA;
        System.out.println("GPA updated to " + newGPA + " for " + name);
    }
    
    // Method to display student information
    public void display() {
        System.out.println("Student: " + name + " (ID: " + id + ", GPA: " + gpa + ")");
    }
    
    // Method to check if student is honor student
    public boolean isHonorStudent() {
        return gpa >= 3.5;
    }
    
    @Override
    public String toString() {
        return "Student{name='" + name + "', id=" + id + ", gpa=" + gpa + "}";
    }
    
    public static void main(String[] args) {
        Student student1 = new Student("Alice Johnson", 12345, 3.8);
        
        student1.display();
        
        if (student1.isHonorStudent()) {
            System.out.println(student1.getName() + " is an honor student!");
        }
        
        student1.setGPA(3.9);
        student1.display();
    }
}

File: ObjectClass/object.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

typedef struct {
    char name[50];
    int id;
    float gpa;
} Student;

// Constructor-like function
Student* createStudent(const char* name, int id, float gpa) {
    Student* student = (Student*)malloc(sizeof(Student));
    strcpy(student->name, name);
    student->id = id;
    student->gpa = gpa;
    return student;
}

// Method-like functions
void displayStudent(const Student* student) {
    printf("Student: %s (ID: %d, GPA: %.2f)\n", 
           student->name, student->id, student->gpa);
}

void updateGPA(Student* student, float newGPA) {
    student->gpa = newGPA;
    printf("GPA updated to %.2f for %s\n", newGPA, student->name);
}

int isHonorStudent(const Student* student) {
    return student->gpa >= 3.5;
}

void destroyStudent(Student* student) {
    free(student);
}

int main() {
    Student* student1 = createStudent("Alice Johnson", 12345, 3.8);
    
    displayStudent(student1);
    
    if (isHonorStudent(student1)) {
        printf("%s is an honor student!\n", student1->name);
    }
    
    updateGPA(student1, 3.9);
    displayStudent(student1);
    
    destroyStudent(student1);
    return 0;
}